### PR TITLE
EE-504 accessioning alpha application

### PR DESCRIPTION
**Pull request by jorizci**
- [x] Study
- [x] Sample (OPEN ACCESS)
- [x] Sample (CONTROLLED ACCESS)
- [x] File
- [x] Run
- [x] Analysis
- [x] Experiment
- [x] Array
- [x] Dac
- [x] Policy
- [x] Dataset

### CODE REVIEW

File: src/main/java/uk/ac/ebi/ega/accession/array/rest/ArrayDTO.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+public class ArrayDTO implements Array {
+
+    @Column(nullable = false)
+    private String submissionAccount;
+
+    @Column(nullable = false)
+    private String alias;
+
+    public ArrayDTO(Array array) {
```
Comment (selvaebi) [inline src/main/java/uk/ac/ebi/ega/accession/array/rest/ArrayDTO.java]: Don't we need a default constructor for all dto's ?

File: src/main/java/uk/ac/ebi/ega/accession/study/persistence/StudyEntityRepository.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ega.accession.study.persistence;
+
+import org.springframework.stereotype.Repository;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.repositories.IAccessionedObjectRepository;
+import uk.ac.ebi.ega.accession.file.persistence.FileEntity;
```
Comment (selvaebi) [inline src/main/java/uk/ac/ebi/ega/accession/study/persistence/StudyEntityRepository.java]: unused import

File: src/main/java/uk/ac/ebi/ega/accession/array/persistence/HistoricArrayEntity.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+package uk.ac.ebi.ega.accession.array.persistence;
+
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.entities.InactiveAccessionEntity;
+import uk.ac.ebi.ega.accession.array.model.Array;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+
+@Entity
+public class HistoricArrayEntity extends InactiveAccessionEntity<Array, Long> implements Array {
```
Comment (selvaebi) [inline src/main/java/uk/ac/ebi/ega/accession/array/persistence/HistoricArrayEntity.java]: Just now noticed , we need entitylisteners for mapped superclass. Historic entity needs created date .

Comment (jorizci) [reply src/main/java/uk/ac/ebi/ega/accession/array/persistence/HistoricArrayEntity.java]: Not needed anymore after adding auditing configuration

**Pull request merged**